pubpub/platform
# Add community ID's to pubfields

## Issue(s) Resolved
Closes https://github.com/pubpub/v7/issues/448

* [x] Adds a new column `communityId` to the pub fields table that cannot be null
* [x] Adds a data migration which tries its best to migrate existing data (needs to be tested more)
* [x] Convert `pubpub:fileUpload` to `unjournal:fileUpload`
* [x] Removes registering core pub fields from the seed script and instrumentation
* [x] Update seed script so that croccroc adds its own pub fields
* [x] Update actions to not use core pub fields
* [x] Updates adding a community to also add its own example pub fields rather than using the core pub fields

## Test Plan
To test the migration:
1. First check out `main` and run a `pnpm reset`. This should get the db into a previous state (hopefully similar to what is in prod right now)
2. Remove all pub values associated with core pub fields (`pubpub:*`). I did this through prisma. If you're doing this from a fresh restart, this will likely leave you with no pub values at all. So it's probably also good to add some pub values by creating a pub in the Unjournal community, which will create pub values on `unjournal:*` fields
3. Now checkout this branch and run `pnpm --filter core run migrate-dev`. It should succeed, and if you query your db you should see fewer pubfields (only `unjournal` ones) and they should have the unjournal community ID 

hooooopefully that's enough to work in prod...

Should also test the push to v6 action, but I'm not sure how to test it.

To test the seed script:
1. On this branch, run a `pnpm reset`. This should create relevant `croccroc` pub fields instead of using the core ones (and should not create the core ones anymore).

Other than that, things that should be fixed now:
* Should only see the fields that belong to the community in the fields page
* Should only see options for fields in the action pubfield selectors
* Also fixed what seemed to be a regression in the email input where you couldn't search for members
* When making a new community, will also create a few pubfields scoped to that community

## Screenshots (if applicable)
![image](https://github.com/user-attachments/assets/acb18f94-06f5-424a-98ec-0eb2636f9a46)
![image](https://github.com/user-attachments/assets/bc4f88a2-17ce-466c-848c-675c1ddab991)
![image](https://github.com/user-attachments/assets/f0290ff1-db72-4082-9eb8-53dcefae6d15)


## Optional

### Notes/Context/Gotchas

### Supporting Docs
